modules/database/src/main/kotlin/dev/proxyfox/database/InMemoryDatabase.kt

```diff
@@ -256,13 +256,13 @@ class InMemoryDatabase : Database() {
         return messages.findLast { it.systemId == systemId && it.channelId == channelId.value }
     }
 
-    override suspend fun getOrCreateTokenFromSystem(systemId: String): TokenRecord {
-        if (!systemTokens.containsKey(systemId)) systemTokens[systemId] = TokenRecord(generateToken(), systemId)
-        return systemTokens[systemId]!!
+    override suspend fun fetchToken(token: String): TokenRecord? {
+        if (!systemTokens.containsKey(token)) return null
+        return systemTokens[token]
     }
```

Comment:
This can be simplified to just `return systemTokens[token]` instead of querying the map twice.